### PR TITLE
Remove the utils package

### PR DESCRIPTION
**Pull request by vdemeester**
This removes the `github.com/docker/docker/utils` package and move code around in *better* places.

* Move `TestDirectory` function (used in a test in `registry` to `pkg/testutil`)
* Move *process* fuction (`IsProcessAlive`, `KillProcess`, …) to `pkg/system`
* Move debug functions to `cli/debug` are they are mainly used for `cli`
* Move `utils/templates` to `pkg/templates`
* move `ReplaceOrAppendEnvValues` to `container` package
* Move names regexp to the `api` package (I think it made sense)

I did separate commits for each of this point but I can squash if required 👼 

/cc @thaJeztah @cpuguy83 @dnephin @icecrime

### CODE REVIEW

File: pkg/testutil/tempfile/tempfile.go [hunk 1]
```diff
@@ -34,3 +39,40 @@ func (f *TempFile) Name() string {
 func (f *TempFile) Remove() {
 	os.Remove(f.Name())
 }
+
+var globalTestID string
```
Comment (AkihiroSuda) [inline pkg/testutil/tempfile/tempfile.go]: why global?

Comment (vdemeester) [inline pkg/testutil/tempfile/tempfile.go]: @AkihiroSuda yeah I just moved the code. I was wondering the same thing :sweat_smile:

**Review by dnephin — APPROVED** [review]
LGTM

Nice change!

File: container/container_windows.go [hunk 2]
```diff
@@ -30,7 +29,7 @@ func (container *Container) CreateDaemonEnvironment(_ bool, linkedEnv []string)
 	// because the env on the container can override certain default values
 	// we need to replace the 'env' keys where they match and append anything
 	// else.
-	return utils.ReplaceOrAppendEnvValues(linkedEnv, container.Config.Env)
+	return ReplaceOrAppendEnvValues(linkedEnv, container.Config.Env)
```
Comment (tiborvass) [inline container/container_windows.go]: this change should not be in this commit.

**Review by tiborvass — CHANGES_REQUESTED** [review]

File: container/container_windows.go (hunk 2 above)
Comment (vdemeester) [inline container/container_windows.go]: oh :scream_cat:

Comment (vdemeester) [inline container/container_windows.go]: Should be in the right commit :angel: :stuck_out_tongue:

File: distribution/registry_unit_test.go [hunk 3]
```diff
@@ -131,3 +135,36 @@ func TestTokenPassThruDifferentHost(t *testing.T) {
 		t.Fatal("Redirect should not forward Authorization header to another host")
 	}
 }
+
+// TestDirectory creates a new temporary directory and returns its path.
```
Comment (thaJeztah) [inline distribution/registry_unit_test.go]: nit: s/TestDirectory/testDirectory/

**Review by thaJeztah — APPROVED** [review]
LGTM, much cleaner!

**Pull request merged**